tests/rptest/services/admin_ops_fuzzer.py

```diff
@@ -523,69 +523,66 @@ def is_deletable(ctx, t):
+
+            deletable_topic = _random_choice(self.prefix, deletable_topics)
+            deletable_partitions = [
+                p for p in ctx.rpk().describe_topic(deletable_topic)
+                if not is_partition_empty(p)
+            ]
+
+            return None if len(deletable_partitions) == 0 else (
+                deletable_topic, random.choice(deletable_partitions))
 
-        topic_partitions = deletable_topic_partitions(ctx)
+        topic_partition = deletable_topic_partition(ctx)
+        if topic_partition is None:
+            return ()
 
         def random_truncate_offset(p):
             return random.randint(
                 int(p.start_offset) + 1, int(p.high_watermark))
 
-        return {
-            topic: {int(p.id): random_truncate_offset(p)
-                    for p in partitions}
-            for topic, partitions in topic_partitions.items()
-        }
+        (topic, partition) = topic_partition
+        return (topic, (int(partition.id), random_truncate_offset(partition)))
 
     def execute(self, ctx):
-        self.truncate_points = self._random_truncate_points(ctx)
-        if len(self.truncate_points) == 0:
+        self.truncate_point = self._random_truncate_point(ctx)
+        if len(self.truncate_point) == 0:
             return False
 
         ctx.redpanda.logger.info(
-            f"Issuing DeleteRecords command: {self.truncate_points}")
-        ctx.kcl().delete_records(self.truncate_points)
+            f"Issuing DeleteRecords command: {self.truncate_point}")
+        (topic, partition_offset) = self.truncate_point
+        (partition, offset) = partition_offset
+        ctx.rpk().trim_prefix(topic, offset, partitions=[partition])
         return True
 
     def validate(self, ctx):
-        if len(self.truncate_points) == 0:
+        if len(self.truncate_point) == 0:
             return False
 
         ctx.redpanda.logger.info(
-            f"Validing topic low watermarks, expecting: {self.truncate_points}"
-        )
+            f"Validing topic low watermarks, expecting: {self.truncate_point}")
 
-        topic_partitions_data = {
-            topic: ctx.rpk().describe_topic(topic)
-            for topic in self.truncate_points.keys()
-        }
+        (topic, partition_offset) = self.truncate_point
+        (partition_id, lwm) = partition_offset
 
-        for topic, partition_lwm in self.truncate_points.items():
-            tpt = topic_partitions_data.get(topic, None)
-            if tpt is None:
-                return False
-            partition_map = {int(p.id): p for p in tpt}
-            for partition, lwm in partition_lwm.items():
-                tpp = partition_map.get(int(partition), None)
-                if tpp is None or int(tpp.start_offset) != lwm:
-                    return False
-        return True
+        partitions_info = ctx.rpk().describe_topic(topic)
+        partition_info = [
+            p for p in partitions_info if int(p.id) == partition_id
+        ]
```

Comment:
nit: maybe worth trying to consume right below and right after to make sure we see the expected errors/offsets?

Reply:
Sorry, merging per offline conversations. But @graphcareful -- you wanna have a follow-up PR for these points?